pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -553,6 +556,53 @@ func DefaultGenerators(cmdName string) map[string]kubectl.Generator {
 	return generator
 }
 
+// fallbackGeneratorNameIfNecessary returns the name of the old generator
+// if server does not support new generator. Otherwise, the
+// generator string is returned unchanged.
+//
+// If the generator name is changed, print a warning message to let the user
+// know.
+func FallbackGeneratorNameIfNecessary(
+	generatorName string,
+	resourcesList []*metav1.APIResourceList,
+	cmdErr io.Writer,
+) string {
+	switch generatorName {
+	case DeploymentBasicAppsV1Beta1GeneratorName:
+		if !Contains(resourcesList, appsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
+
+			fmt.Fprintf(cmdErr,
```

Comment:
If you parameterize the API you can turn this into a single utility method for printing the error and use it below. That will clean up this code a bunch.